magenta-lib/src/main/scala/magenta/deployment_type/S3.scala

```diff
@@ -109,32 +109,32 @@ object S3 extends DeploymentType with S3AclParams {
   }
 
   def perAppActions = {
-    case "uploadStaticFiles" => (pkg) => (reporter, resourceLookup, parameters, stack) => {
+    case "uploadStaticFiles" => (pkg) => (runtime) => {
       def resourceLookupFor(resource: Param[String]): Option[Datum] = {
         resource.get(pkg).flatMap { resourceName =>
           assert(pkg.apps.size == 1, s"The $name package type, in conjunction with ${resource.name}, only be used when exactly one app is specified - you have [${pkg.apps.map(_.name).mkString(",")}]")
-          val dataLookup = resourceLookup.data
+          val dataLookup = runtime.lookup.data
           val app = pkg.apps.head
-          val datumOpt = dataLookup.datum(resourceName, app, parameters.stage, stack)
+          val datumOpt = dataLookup.datum(resourceName, app, runtime.parameters.stage, runtime.stack)
```

Comment:
To attempt to bolster my earlier point, there are lots of places like this where DeployParameters and Stack appear together without the Lookup and Reporter.